cmd/lifecycle/main.go

```diff
@@ -18,31 +18,41 @@ import (
 )
 
 func main() {
-	switch strings.TrimSuffix(filepath.Base(os.Args[0]), filepath.Ext(os.Args[0])) {
+	phase := strings.TrimSuffix(filepath.Base(os.Args[0]), filepath.Ext(os.Args[0]))
```

Comment:
fwiw I'm a fan of explicitly naming this string `phase` instead of leaving it anonymous 👏